renku/core/dataset/providers/renku.py

```diff
@@ -388,10 +367,26 @@ def _fetch_dataset(self, client_dispatcher: IClientDispatcher, database_dispatch
             self._migrate_project()
             self._project_repo = repository
 
-            dataset = DatasetsProvenance().get_by_name(self._name)
+            datasets_provenance = DatasetsProvenance()
+
+            dataset = datasets_provenance.get_by_name(self._name)
             if not dataset:
                 raise errors.ParameterError(f"Cannot find dataset '{self._name}' in project '{self._project_url}'")
+
+            if self._tag:
+                tags = datasets_provenance.get_all_tags(dataset=dataset)
+                tag = next((t for t in tags if t.name == self._tag), None)
+
+                if tag is None:
+                    raise errors.ParameterError(f"Cannot find tag '{self._tag}' for dataset '{self._name}'")
+
+                dataset = datasets_provenance.get_by_id(tag.dataset_id.value)
+
+            assert dataset is not None
             provider_dataset = ProviderDataset.from_dataset(dataset)
+
+            # NOTE: Set the dataset version so that it can be checked later to see if a tag was specified for import
+            provider_dataset.version = self._tag
```

Comment:
There is code on import to create a tag if `.version` is set. Would this result in duplicate tags being created?

Reply:
But it has a different description, "Tag generated by import"

Reply:
I'm not sure what's the right thing to do here: On one hand we lose the description of the original tag (if there is one) and on the other hand we lose the description that says this tag was imported. I went with this approach since we have the original tag name. I can change it to keep the original tag as well. What do you think?